pkg/controller/containersource/resources/deployment.go

```diff
@@ -79,3 +81,14 @@ func MakeDeployment(org *appsv1.Deployment, args *ContainerArguments) *appsv1.De
 
 	return deploy
 }
+
+func sinkArg(args *ContainerArguments) string {
+	if args.SinkInArgs {
+		for _, a := range args.Args {
+			if strings.HasPrefix(a, "--sink=") {
+				return strings.Replace(a, "--sink=", "", -1)
```

Comment:
Good catch!

Reply:
Weird when the tests help you find an issue!!